kitsune/search/v2/search.py

```diff
@@ -0,0 +1,209 @@
+
+    @property
+    @abstractmethod
+    def index(self):
+        pass
+
+    @property
+    @abstractmethod
+    def fields(self):
+        """Array of fields to search over."""
+        pass
+
+    @property
+    @abstractmethod
+    def highlight_fields(self):
+        """Array of fields to highlight."""
+        pass
+
+    @property
+    def filter(self):
+        """Filter which returns all documents of this type to search over."""
+        return Q("match", _index=self.index)
+
+    @abstractmethod
+    def product_filter(self, product):
+        """Filter which returns documents with the correct product."""
+        pass
+
+    @abstractmethod
+    def make_result(self, hit):
+        """Transform a hit into a result dictionary."""
+        pass
+
+    @staticmethod
+    def extract_highlight(hit):
+        highlight = getattr(hit.meta, "highlight", None)
+        if highlight:
+            return bleach.clean(
+                next(iter(highlight.to_dict().values()))[0],
+                tags=["strong"],
+                strip=True,
+            )
+        return None
+
+
+class AAQConfig(BaseConfig):
+    """Search config for AAQ."""
+
+    index = QUESTION_INDEX_NAME
+    fields = [
+        "question_title.{}^4",
+        "question_content.{}^3",
+        "answer_content.{}^3",
+    ]
+    highlight_fields = ["question_content.{}", "answer_content.{}"]
+    filter = Q("bool", must_not=Q("exists", field="updated"))
+
+    def product_filter(self, product):
+        return Q("match", question_product_id=product.id)
+
+    def make_result(self, hit):
+        locale = hit.locale
+
+        summary = self.extract_highlight(hit)
+        if not summary:
+            summary = bleach.clean(hit.question_content[locale][:500], strip=True)
+
+        num_answers = 0
+        answer_content = getattr(hit, "answer_content", None)
+        if answer_content:
+            num_answers = len(answer_content[locale])
+
+        return {
+            "type": "question",
+            "url": "/questions/{}".format(hit.question_id),
+            "score": hit.meta.score,
+            "title": hit.question_title[locale],
+            "search_summary": summary,
+            "last_updated": datetime.fromisoformat(hit.question_updated),
+            "is_solved": hit.question_has_solution,
+            "num_answers": num_answers,
+            "num_votes": hit.question_num_votes,
+        }
+
+
+class KBConfig(BaseConfig):
+    """Search config for the Knowledge Base."""
+
+    index = WIKI_DOCUMENT_INDEX_NAME
+    fields = [
+        "keywords^8",
+        "title.{}^6",
+        "summary.{}^2",
+        "content.{}^1",
+    ]
+    highlight_fields = ["summary.{}", "content.{}"]
+
+    def product_filter(self, product):
+        return Q("match", product_ids=product.id)
+
+    def make_result(self, hit):
+        locale = hit.locale
+
+        summary = self.extract_highlight(hit)
+        if not summary:
```

Comment:
I would rewrite this focusing first on `hit.summary[locale]` and `hit.content`. You can determine which of the 2 exists and then maybe run it through bleach. That should remove the need for two identical `if`

Reply:
This has changed a little bit to handle the case where a document has no summary attribute, but imo the double `if not summary` is the clearest way of expressing this: if we haven't got a summary yet, do this, if we still haven't got a summary after that, do this other thing.